base/v0_5_exp/schema.go

```diff
@@ -84,6 +84,11 @@ type FilesystemOption string
 
 type Group string
 
+type Grub struct {
+	Inline *string `yaml:"inline"`
+	Local  *string `yaml:"local"`
```

Comment:
The user could just declare the `/boot` filesystem and the `user.cfg` contents directly, right?  If all we want to do is let users provide an arbitrary `user.cfg`, we'd normally document an example config, rather than implementing sugar.

I do think sugar would be useful here, but it should be at a higher semantic level.  Rather than exposing GRUB config syntax, we could provide fields for specific configuration knobs, similar to the existing `PasswdUser` struct.  That also helps discourage users from trying to arbitrarily change the bootloader configuration, which we surely don't want to support.  Is there a specific list of features/commands we want to enable?

Reply:
The only one we need to enable currently is GRUB password support. Other than that, I haven't tinkered much with GRUB configurations so I'm not sure what some nice-to-haves would be.